pkg/bgp/speaker/k8s.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package speaker
+
+import (
+	"context"
+
+	"github.com/cilium/cilium/pkg/k8s"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// client wraps the K8s client so that it can conform to the client from
+// MetalLB.
+type client struct {
+	*k8s.K8sClient
+}
+
+func (c *client) Update(svc *v1.Service) (*v1.Service, error) {
+	return c.CoreV1().Services(svc.GetNamespace()).Update(context.TODO(), svc, metav1.UpdateOptions{})
+}
+func (c *client) UpdateStatus(svc *v1.Service) error {
+	_, err := c.CoreV1().Services(svc.GetNamespace()).UpdateStatus(context.TODO(), svc, metav1.UpdateOptions{})
+	return err
+}
+
+func (*client) Infof(_ *v1.Service, desc, fmt string, args ...interface{}) {
+	log.WithField("event", desc).Infof(fmt, args...)
+}
+func (*client) Errorf(_ *v1.Service, desc, fmt string, args ...interface{}) {
+	log.WithField("event", desc).Errorf(fmt, args...)
+}
```

Comment:
I initially kept it as duplicated code because it was such a small amount of code, but then again, as we expand on BGP, this package might evolve, so it makes sense to refactor this to a common place for future-proofing.